env/export.go

```diff
@@ -87,40 +101,61 @@ func FromExport(body string) Environment {
 				continue
 			}
 
-			// Trim the `declare -x ` off the start of the line
-			line = strings.TrimPrefix(line, "declare -x ")
+			if strings.HasPrefix(line, "declare") {
+				command := strings.Split(line, " ")
+				if len(command) < 3 {
+					// There should be at least three elements to the command; the format is:
+					// declare -x VARNAME="VALUE"
+					// If there's anything less than that, something is seriously wonky with this line and we won't be able to parse it anyway, so just skip it
+					continue
+				}
+
+				// Massive assumption here! We're assuming that that the options passed to declare have been coalesced -
+				// ie, that the command looks like `declare -axtn VARNAME="BLARGH"` rather than `declare -x -a -t -n VARNAME="BLARGH"`
+				// However, the output of `export -p` (which is what we're parsing here) is pretty consistent in coalescing the args
+				// God willing, this shouldn't change.
```

Comment:
he says, tempting fate

Reply:
Heh, this crossed my mind too.
I think if we ran into that, it would be time to explore a non-`export -p`-based solution 😅